src/header.rs

```diff
@@ -1540,17 +1556,29 @@ fn copy_into(slot: &mut [u8], bytes: &[u8]) -> io::Result<()> {
 /// * a nul byte was found
 /// * an invalid path component is encountered (e.g. a root path or parent dir)
 /// * the path itself is empty
-fn copy_path_into(mut slot: &mut [u8], path: &Path, is_link_name: bool) -> io::Result<()> {
+fn copy_path_into(
+    mut slot: &mut [u8],
+    path: &Path,
+    is_link_name: bool,
+    is_truncated_gnu_long_path: bool,
```

Comment:
This is OK as is but...I think functions with more than one `bool` parameter can be hard to read on the call side. From a quick glance they seem to be mutually exclusive? We could add a `copy_path_into_inner` that takes both bools, but have a thin wrapper like

```
fn copy_path_into_gnu_long(...) {
    copy_path_into_impl(...)
}
```
or so?

Reply:
This will make it easier to use the vanilla `copy_path_into` without considering the need for this setting - but I understand that in most cases that's the right choice so let's do it.
If it becomes a problem perhaps a better way would be to pass some option struct containing all the `bools` with settings.